### DISCUSSION TITLE
How to detect and use the page language in nextJS

### DISCUSSION THREAD

**Discussion by lucasReghiniM** [vercel/next.js#20210] [Help]
Im trying to make a site that could identify the country and set the language.
that is the first thing that i tried:

```
import * as pt from "./pt";
import * as en from './en';

let translate;

if (window.navigator.language == 'pt-PT' || window.navigator.language == 'pt-BR') {
  translate = pt.default;
} else {
  translate = en.default;
}
export default translate
```
the pt/en files its just JSONS

and i use like this:

`{Translate.header.label_opts_clients}
`

But window doesnt exist in nextJS
That error appears: ReferenceError: window is not defined

I need to use this in a XML (react) file and in a JSON file 
like:

```
export const headerOptions = [
  {
    title: "`${Translate.header.label_opts_home}`",
...
```

How can i do this?
Cause i know that exist useRouter(hook), so if you need to take the url in a page you can use like this:

```
...

const { pathname } = useRouter()
  
  const [crumbs, setCrumbs] = useState<string[]>();

  useEffect(()=>{
    renderCrumbs()
  },[pathname])

  const setHref = (links:string[], index:number|string) => {
    let linkPath = ""
    
    for (let i = 0; i <= index; i++) {      
      linkPath = linkPath + "/" + links[i]      
    }

    return linkPath
  }

  const renderCrumbs = () => {
    let links = pathname.split('/').slice(1);
    let limit = links.length;
    let array = [];

...

```
But you cant use hooks out of a function.

i tried to put the condition in a function and return the let translate, and use the translate with parentheses ({Translate().header.something}) but didnt work.

**Comment by stigkj** [comment]
The reason for the `ReferenceError: window is not defined` error is that Next.js renders the pages on the server where `window` does not exist.

Next.js has built-in support for locale detection that can be accessed from the `router` object, tl;dr: `router.locale`. For more information: https://nextjs.org/docs/advanced-features/i18n-routing#accessing-the-locale-information
